src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java

```diff
@@ -1,23 +1,111 @@
 // I'm bad at naming things. Please come up with a better name...
 package com.spartronics4915.frc2020.subsystems;
 
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.ColorSensorV3;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.spartronics4915.frc2020.Constants;
 import com.spartronics4915.lib.subsystems.SpartronicsSubsystem;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.DigitalInput;
 
 public class PanelRotator extends SpartronicsSubsystem {
+    private final CANSparkMax mSpinMotor;
+    private final CANSparkMax mExtendMotor;
+    
+    private final DigitalInput mBeamSensorUp;
+    private final DigitalInput mBeamSensorDown;
+    
+    private final ColorSensorV3  mColorSensor;
+
+    //TODO: These are essentially random numbers, with the max value based on the images at
+    // https://www.andymark.com/products/infinite-recharge-control-panel-stickr
+    public int[] mMinimumRed = {200, 0, 0};
+    public int[] mMaximumRed = {255, 30, 30};
+
+    //TODO: These are bad and will work in a way that will make you lose, which will be sad
+    public int[] mMinimumGreen = {0, 200, 0};
+    public int[] mMaximumGreen = {30, 255, 30};
+
+    //TODO: These are bad and will work in a way that will make you lose, which will be sad
+    public int[] mMinimumBlue = {0, 200, 200};
+    public int[] mMaximumBlue = {30, 255, 255};
+
+    //TODO: These are bad and will work in a way that will make you lose, which will be sad
+    public int[] mMinimumYellow = {200, 200, 0};
+    public int[] mMaximumYellow = {255, 255, 30};
+
+    public String sensedColor;
+    
+    public int red;
+    public int green;
+    public int blue;
 
     public PanelRotator() {
-        
+        mBeamSensorUp = new DigitalInput(Constants.kPanelRotatorBeamSensorUpID);
+        mBeamSensorDown = new DigitalInput(Constants.kPanelRotatorBeamSensorDownID);
+        mSpinMotor = new CANSparkMax(Constants.kPanelRotatorSpinMotorID, MotorType.kBrushless);
+        mExtendMotor = new CANSparkMax(Constants.kPanelRotatorExtendMotorID, MotorType.kBrushless);
+        mColorSensor = new ColorSensorV3(I2C.Port.kOnboard);
     }
 
-    public void rotate(double number) {
+    public void raise() {
+        mExtendMotor.set(Constants.kPanelRotatorExtendMotorSpeed);
+        if(!mBeamSensorUp.get())
+            mExtendMotor.set(0);
+    }
 
```

Comment:
We still don't have encoder specifications from Design - so for now, create `getBeamSensorUp()` and `getBeamSensorDown()` methods that return a boolean `mBeamSensorUp/Down.get()`.